### PR TITLE
Luong Attention

### PR DESCRIPTION
**Pull request by ArnoldLIULJ**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Based on https://arxiv.org/pdf/1508.04025.pdf
ADD attention-based seq2seq model with 3 alternatives to calculate the attention score.
ADD tests for attention-based seq2seq model

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'tensorlayer'
+copyright = '2019, lingjun liu'
+author = 'lingjun liu'
```
Comment (JingqingZ) [inline conf.py]: @zsdonghao Could you have a check if this file is corrected written?

**Comment by ArnoldLIULJ** [comment]
- Rebase done
- Add __init__  and __ALL__ done
- Modify docs
- Unit-test based on sin-sequence with Gaussian distribution noise. However, performance not convincing. Need your double check.
- Re-test by running the chatbot, generating better results compared to simple multi-stacked Seq2Seq

**Comment by JingqingZ** [comment]
> * Rebase done
> * Add **init**  and **ALL** done
> * Modify docs
> * Unit-test based on sin-sequence with Gaussian distribution noise. However, performance not convincing. Need your double check.
> * Re-test by running the chatbot, generating better results compared to simple multi-stacked Seq2Seq

I will double check the unittest.

BTW: After this PR is merged, you may upload your code which uses Luong attention to the chat-bot repo as an example of using Luong attention.

File: conf.py (hunk 1 above)
Comment (zsdonghao) [reply conf.py]: What is this file for??? There exists `docs/conf.py` for the docs template.
I think this file is duplicated.

File: docs/modules/models.rst [hunk 2]
```diff
@@ -31,9 +34,20 @@ VGG19
...
-.. autofunction:: MobileNetV1
+.. autoclass:: MobileNetV1
+
+Seq2seq
+------------------------
+
+.. autoclass:: Seq2seq
+
+
+Seq2seq_Luong_Attention
```
Comment (zsdonghao) [inline docs/modules/models.rst]: AAA_AAA --> AAA AAA

File: index.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
+.. tensorlayer documentation master file, created by
```
Comment (zsdonghao) [inline index.rst]: do you mean `docs/index.rst` ??

**Comment by ArnoldLIULJ** [comment]
- Rewrite sin sequence with noise testing. 
- Remove conf.py and index.rst

File: index.rst (hunk 3 above)
Comment (ArnoldLIULJ) [reply index.rst]: I have already removed index.rst and confirm.py

**Comment by JingqingZ** [comment]
I think `make.bat` should be deleted as well as it is already in the `docs/` folder.

Have you tested if the attention can learn the period of sin function?

**Comment by ArnoldLIULJ** [comment]
> I think `make.bat` should be deleted as well as it is already in the `docs/` folder.
> 
> Have you tested if the attention can learn the period of sin function?

Yes. It can learn the sequence.

**Comment by zsdonghao** [comment]
`Seq2seqLuongAttention` is better for Python naming

**Comment by ArnoldLIULJ** [comment]
Change model name to Seq2seqLuongAttention
Modify changelog.md

**Comment by JingqingZ** [comment]
@zsdonghao Please check this branch and merge when you feel it is ready. And how to solve the Travis-CI problem?

**Comment by ArnoldLIULJ** [comment]
I will update the chatbot repository with an attention version when this merge is done.
Or you are suggesting add an example of chatbot in the \examples folder of tensorlayer?

**Comment by JingqingZ** [comment]
> I will update the chatbot repository with an attention version when this merge is done.
> Or you are suggesting add an example of chatbot in the \examples folder of tensorlayer?

A chatbot example in the chatbot repo would be fine.

**Comment by ArnoldLIULJ** [comment]
Sure. I will do it after the mergence

**Pull request closed without merging**